core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -227,10 +229,11 @@ protected void doClose() {
         keepAliveReaper.cancel();
     }
 
-    public DfsSearchResult executeDfsPhase(ShardSearchRequest request) throws IOException {
+    public DfsSearchResult executeDfsPhase(ShardSearchRequest request, SearchTask task) throws IOException {
         final SearchContext context = createAndPutContext(request);
```

Comment:
See @nik9000's comment bellow, this doesn't seem to be a good way to handle it since the task can change over time.